alf/utils/common.py

```diff
@@ -1449,7 +1449,14 @@ def get_all_parameters(obj):
             # when called in a wrong context (e.g. Algorithm.experience_spec)
             if isinstance(getattr(type(obj), name, None), property):
                 continue
-            attr = getattr(obj, name, None)
+            attr = None
+            try:
+                attr = getattr(obj, name)
+            except:
+                # some attrbutes are property function, which may raise exception
+                # when called in a wrong context (e.g. Algorithm.experience_spec)
```

Comment:
Property function  is handled by line 1450.
This is for handling the case that the attribute cannot be accessed.

Reply:
Oh yes, updated.